ui/lib/kv/addon/controllers/secret/details/index.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Controller from '@ember/controller';
+import Ember from 'ember';
+import { tracked } from '@glimmer/tracking';
+import { task, timeout } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+
+export default class KvSecretDetailsIndexController extends Controller {
+  @service store; // this.store is referenced in this.fetchSyncStatus()
+
+  @tracked syncDestinations;
+
+  // task is cancelled by resetController() upon leaving the kv v2 details route
+  @task
+  *pollSyncStatus() {
+    while (true) {
+      if (Ember.testing) return;
+
+      yield timeout(10000);
+      try {
+        this.syncDestinations = yield this.fetchSyncStatus(this.model);
+      } catch (e) {
+        // otherwise keep polling
+      }
+    }
```

Comment:
Curious if we need to poll? Was that something that came from design? Also where is the `fetchSyncStatus` method coming from?

Reply:
Below I mentioned this as a question I had - we poll tidy status in PKI and this seemed to be similar as it's a status. But I'm not sure how much of a benefit we'll get from polling so I want to confirm with them